ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java

```diff
@@ -96,19 +108,31 @@ int actualPort() {
   private Router setupRouter() {
     final Router router = Router.router(vertx);
 
-    router.route().failureHandler(ServerVerticle::handleFailure);
+    // We only want to apply failure handler and auth handlers to routes that are not proxied
+    // as Jetty will do it's own auth and failure handler
+    routeFailureToNewApi(router, ServerVerticle::handleFailure);
+
+    final Optional<AuthHandler> authHandler = getAuthHandler(server);
+    final KsqlSecurityExtension securityExtension = server.getSecurityExtension();
+    final Optional<AuthenticationPlugin> authenticationPlugin = server.getAuthenticationPlugin();
 
-    getAuthHandler(server).ifPresent(authHandler -> {
-      router.route().handler(ServerVerticle::pauseHandler);
-      router.route().handler(authHandler);
-      server
-          .getSecurityExtension()
-          .getAuthorizationProvider()
-          .ifPresent(ksqlAuthorizationProvider -> router.route()
-              .handler(new KsqlAuthorizationFilter(server.getWorkerExecutor(),
+    if (authHandler.isPresent() || authenticationPlugin.isPresent()) {
+      routeToNewApi(router, ServerVerticle::pauseHandler);
+      if (authenticationPlugin.isPresent()) {
+        // Authentication plugin has precedence
```

Comment:
I am assuming it's the same otherwise, on cloud with current KSQL a user could override the cloud based oauth authentication by providing setting a an auth method in their app config?